torch/csrc/distributed/rpc/utils.cpp

```diff
@@ -355,6 +355,101 @@ std::pair<std::vector<char>, std::vector<at::Tensor>> wireDeserialize(
   return {std::move(payload), std::move(tensors)};
 }
 
+TensorPipeEntry tensorpipeSerialize(const Message& rpcMessage) {
+  tensorpipe::Message tpMessage;
+  std::vector<torch::Tensor> reservedTensors;
+  std::vector<std::vector<uint8_t>> copiedTensors;
+
+  const std::vector<char>& payload = rpcMessage.payload();
+  c10::List<at::Tensor> tensors = cloneSparseTensors(rpcMessage.tensors());
+
+  // Payload
+  tpMessage.data = (uint8_t*)(payload.data());
+  tpMessage.length = payload.size();
+
+  // Metadata - encode rpc message type + message id
+  tpMessage.metadata = std::to_string(static_cast<int>(rpcMessage.type())) +
+      ":" + std::to_string(static_cast<int>(rpcMessage.id()));
+
+  // Tensors
+  tpMessage.tensors.reserve(tensors.size());
+  for (at::Tensor tensor : tensors) {
+    // Keep original user tensors and cloned sparse tensors
+    reservedTensors.push_back(tensor);
+    tensorpipe::Message::Tensor tpTensor;
+    torch::jit::Pickler pickler(
+        [&](const void* buf, size_t sz) -> size_t {
+          tpTensor.metadata.append(static_cast<const char*>(buf), sz);
+          return sz;
+        },
+        nullptr);
```

Comment:
Following D20749423, this `nullptr` should be removed.

Reply:
Yes, it's removed from the 1st diff in stack.. Don't know why it's not updated here